contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpSpanCustomizer.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http;
+
+import io.opencensus.contrib.http.util.HttpTraceUtil;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.SpanBuilder;
+import javax.annotation.Nullable;
+
+/**
+ * A helper class to customize an HTTP span.
+ *
+ * <p>It contains basic instrumentation implementation. See <a
+ * href="https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md">HTTP
+ * trace spec</a> for more information.
+ *
+ * <p>Users can extend this helper to customize builder and end status.
+ *
+ * @param <Q> the HTTP request entity.
+ * @param <P> the HTTP response entity.
+ * @since 0.13
+ */
+public class HttpSpanCustomizer<Q, P> {
+
+  /**
+   * Returns customized span name according to the request.
+   *
+   * <p>Note that in OpenCensus, {@link Span} is immutable and cannot be renamed.
+   *
+   * @param request the HTTP request entity.
+   * @param extractor the framework specific extractor to get information from the request.
+   * @return a string that represents the name of the span.
+   */
+  public String getSpanName(Q request, HttpExtractor<Q, P> extractor) {
+    // default span name
+    return "/" + extractor.getPath(request);
```

Comment:
how can you be sure that the path does not already start with a leading slash? In this case we would have a double `//` in the span name. E.g. in the Akka-Http framework `req.uri.path.toString()` returns the path with a leading slash.

Maybe it makes sense to have a look if `extractor.getPath(request)` already contains a leading slash?

Reply:
Updated.